File: app/schemas/parent_grades.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ParentGradeOut(BaseModel):
    submission_id: int
    assignment_id: int
    student_id: int
    assignment_title: str
    max_score: int
    score: int | None
    feedback: str | None
    status: str
    submitted_at: datetime
    graded_at: datetime | None

    model_config = ConfigDict(from_attributes=True)

File: mhike-school-web/lib/services/parentGrades.ts
import { apiGet } from "@/lib/api";

export type ParentGrade = {
    submission_id: number;
    assignment_id: number;
    student_id: number;
    assignment_title: string;
    max_score: number;
    score: number | null;
    feedback: string | null;
    status: string;
    submitted_at: string;
    graded_at: string | null;
};

export async function getParentGrades(): Promise<ParentGrade[]> {
    return apiGet<ParentGrade[]>(
        "/assignment-submissions/parent/grades",
    );
}